src/contracts/aludel/AludelV3.sol

```diff
@@ -753,6 +753,9 @@ contract AludelV3 is IAludelV3, Ownable, Initializable, Powered {
                 stakes[computedStakeIndex].amount -= amounts[metaIndex];
                 currentStake.amount -= amounts[metaIndex];
             }
+            if (address(_aludel.hookContract) != address(0)) {
+                _aludel.hookContract.unstakeAndClaimPost(currentStake);
+            }
```

Comment:
We could have a `virtual internal` function overrided in AludelV3.sol with these lines, and replace these lines with a call to something like `_unstakeAndClaimPost`.

We're adding an external call for each stake. That'll pile a lot of gas.

Reply:
I think we've discussed this offline and came to an agreement to go for this implementation 🙃 but having this functionality in an external contract would allow us to deploy a single template for AludelV3 and then having users deploy contracts with different functionality without further permissioning on the AludelFactory.

The implementation currently proposed by this PR, especially if we add it to other methods, would enable other people to write extensions without our permissioning. I can think of other modules like whitelists or caps to a stake's size, but perhaps the community could come up with things we haven't tought about.

I mean, I'm okay with doing this as an internal call to save gas, but in that case we should also accept the following compromises as well:

- [ ] increased management cost, due to having to whitelist more aludels in the AludelFactory.
- [ ] increased frontend work, since each different Aludel template would potentially have different parameters, instead of just having a hook contract address for all of them.

(@joehquak probably can update on how much of a nuisance the second item would be)

Reply:
My second comment was only trying to highlight the fact that we're gonna spend more gas on this call.

`unstake` reverts if at least one of all the stakes' timestamps is not valid. Maybe we could just keep the oldest Stake in memory and call `_unstakeAndClaimPost` using the oldest stake the function's behavior it would be almost identical.